### PR TITLE
Add function check_dmesg_output

### PR DESCRIPTION
**Pull request by chloerh**
Allow users to check whether dmesg output contains certain pattern.

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
JOB LOG    : /root/avocado/job-results/job-2021-04-25T16.22-abf8fce/job.log
 (1/1) type_specific.local.memory_hotplug.report_failure: PASS (27.41 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 27.69 s

**Comment by chloerh** [comment]
@Yingshun Help help~

File: virttest/utils_sys.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
+Virtualization test utility functions.
+
+:copyright: 2008-2009 Red Hat Inc.
```
Comment (Yingshun) [inline virttest/utils_sys.py:4]: Can you please update it to the latest? Others look good to me.

Comment (chloerh) [reply virttest/utils_sys.py]: Sure

**Review by Yingshun — APPROVED** [review]

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by pevogam** [comment]
Why isn't this placed in utils_misc next to `utils_misc.verify_dmesg()`? It could even be integrated with the functionality there which could reuse some of the pattern matching.

**Comment by pevogam** [comment]
Also, why can't it immediately reuse or extend `avocado.utils.dmesg`? To me it seems like yet another module raising complexity.

**Comment by chloerh** [comment]
@pevogam Thank u for bringing this useful module to us. But sadly we're still running our test with lts branch, and I have checked that the commit of dmesg module is not included in 82lts. But it still could be very useful to us after we switch to the next lts avocado. Thanks again.

**Comment by pevogam** [comment]
Yes that's understandable, we also use the Avocado 82 LTS because of the tii many changes happening there all the time. Then would you be willing to commit to later migrating the dmesg functionality or dropping it in favor of the upstream one?

**Comment by chloerh** [comment]
@pevogam For future tests, I would recommend dmesg utils of avocado. 
Sorry but I don't quite understand what you meant by 'commit to later migrating the dmesg functionality' :sweat_smile:  Did you mean migrate from avocado master to 82lts?

**Comment by pevogam** [comment]
No I mean would you want to migrate the functionality to Avocado if it extends what is already available there or if not add a TODO in the current function's docstring that is should be dropped in the future with the next LTS and a function pointer there?

**Comment by chloerh** [comment]
> No I mean would you want to migrate the functionality to Avocado if it extends what is already available there or if not add a TODO in the current function's docstring that is should be dropped in the future with the next LTS and a function pointer there?

@pevogam I'd prefer the second one, since we're currently running with 82lts and the function is required by one of our tests.

**Comment by pevogam** [comment]
Sounds good enough, a TODO will be good then so that we keep track of such reimplementations.

**Pull request merged**